tfx/proto/transform.proto

```diff
@@ -0,0 +1,27 @@
+// Copyright 2019 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+syntax = "proto3";
+
+package tfx.components.transform;
+
+// Defines the splits config.
+message SplitsConfig {
+  // Splits that should be analyzed and used to produce the metadata.
+  // hint: this is normally the train dataset.
+  repeated string analyze_splits = 1;
```

Comment:
Any reason why this shouldn't be analyze_and_transform? Do we expect a case where a split is only used for analysis and it shouldn't be transformed?

Reply:
i used analyze_splits and transform_splits to correspond to labels.ANALYZE_DATA_PATHS_LABEL and labels.TRANSFORM_DATA_PATHS_LABEL in the executor.

if we use analyze_and_transform, then we need to do additional work to combine analyze_and_transform and transform_splits for the labels.TRANSFORM_DATA_PATHS_LABEL.

Reply:
@rose-rong-liu

Reply:
I think this is a question of API as opposed to component logic, since the alternative logic would have the same level of complexity, and this is what we currently do in TFX.
So I think that the API question is, do we want users to explicitly specify analyze and transform splits separately? Are there any cases where the user would like to analyze a split, but not transform it (if they're transforming anything)?

```
  splits_config {
    analyze_splits: "train1"
    analyze_splits: "train2"
    transform_splits: "train1"
    transform_splits: "train2"
    transform_splits: "eval"
  }
```
vs.
```
  splits_config {
    analyze_and_transform: "train1"
    analyze_and_transform: "train2"
    transform_splits: "eval"
  }
```

I'd be fine with either option, just want to make sure that we're making a conscious choice here (@rose-rong-liu feel free to chime in).

Also, could we rename analyze_splits/transform_splits->analyze/transform? I think that would be cleaner.

Reply:
In fact we started with analyze_and_transform initially.
Then chenkai's cl/266415037 (columnar support) separated it up, which also changes the implementation of internal transform/runner.py.
I think either way works. If we switch it back, we also need to change transform/runner.py.

Reply:
another thought is if we turn off materialization for transformed example, then set up analyze_and_transform might be a little bit weird.

Re Rose, for the label I think keep then separate is fine, just for the config:
-  analyze_and_transform/transform means the config is shorter as Zohar's example above.
-  analyze/transform is more accurate if materialization is off.

Either is fine to me.

Reply:
I renamed analyze_splits/transform_splits to analyze/transform.

Reply:
Keeping analyze/transform is fine by me.

Reply:
Great, thank you!